File: csharp/lucians-luscious-lasagna/LuciansLusciousLasagna.cs
public class Lasagna
{
    public int ExpectedMinutesInOven() => 40;

    public int RemainingMinutesInOven(int minutesInOven)
    {
        return ExpectedMinutesInOven() - minutesInOven;
    }

    public int PreparationTimeInMinutes(int layerCount)
    {
        const int LayerPreparationTimeInMinutes = 2;
        return layerCount * LayerPreparationTimeInMinutes;
    }

    public int ElapsedTimeInMinutes(int layerCount, int minutesInOven)
    {
        return PreparationTimeInMinutes(layerCount) + minutesInOven;
    }
}

File: fsharp/lucians-luscious-lasagna/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let remainingMinutesInOven minutesInOven = expectedMinutesInOven - minutesInOven

let preparationTimeInMinutes numberOfLayers =
    let minutesPerLayer = 2
    minutesPerLayer * numberOfLayers

let elapsedTimeInMinutes numberOfLayers minutesInOven =
    preparationTimeInMinutes numberOfLayers + minutesInOven